[elasticsearch] test body quantization by indexing a real document

pawelchcki opened the pull request:
done to test if the quantization changes made for mongo did not affect ES body quantization too much. 

Todo: 
- [ ] release 0.13 and rebase this branch on master

delner commented on Appraisals:
```diff
@@ -62,7 +62,7 @@ elsif Gem::Version.new('1.9.3') <= Gem::Version.new(RUBY_VERSION) \
       gem 'aws-sdk', '~> 2.0'
       gem 'dalli'
-      gem 'elasticsearch-transport'
+      gem 'elasticsearch'
```
Why did we change this dependency?

pawelchcki replied:
`elasticsearch` gem contains `elasticsearch-transport` https://github.com/elastic/elasticsearch-ruby/blob/master/elasticsearch/elasticsearch.gemspec#L25

And it provides much higher level API allowing easy use of ES documentation examples as bases for tests.

e.g.
```subject(:request) { client.cluster.health }```
or 
```subject(:request) { client.index index: index_name, type: document_type, id: document_id, body: document_body }```

Before we were using lower level api like:
```client.perform_request 'GET', '_cluster/health'```

delner replied:
I don't think we want to broaden our support to `elasticsearch` as a whole unless there's a strong reason for doing so. Is there a reason why we need to broaden this to support the change you're making? Is there not an analogous way to do so with just `elasticsearch-transport`? If so, how is that the case?

pawelchcki replied:
I've been wary of using low lever API of `elasticsearch-transport` to avoid making the tests understand too much of ES internals. However since we are already tagging relatively low level things by using ES url etc, so we're already quite deep into ES internals.

I've switched the test to only `elasticsearch-transport` to as you pointed out avoid broadening of the support to another gem.

pawelchcki commented:
@delner Thanks for the review! I've switched back to using old gem. However I also added one additional test case to mimick the  rest of old ES test I based this spec off of. So that we can retire the old test case - which I also did.

delner commented on spec/ddtrace/contrib/elasticsearch/patcher_spec.rb:
```diff
@@ -0,0 +1,121 @@
+require 'spec_helper'
```
These specs are not running in CI. You need to add these specs to the [appropriate spots in the Rakefile](https://github.com/DataDog/dd-trace-rb/blob/965809a782dc3c5dbb6fcbd4a004db17b663ac2a/Rakefile#L225).

delner commented on test/contrib/elasticsearch/miniapp_test.rb:
```diff
@@ -1,66 +0,0 @@
-require 'time'
```
💯

delner commented on spec/support/http_helpers.rb:
```diff
@@ -21,4 +24,18 @@ def mock_http_request(options = {})
+        res = Net::HTTP.get_response(uri)
+        return true if res.code == '200'
+      rescue StandardError => e
+        puts e if i >= 3 # display errors only when failing repeatedly
```
Using `puts` to signal problems doesn't seem sufficient. What's the intent here?

If we'd like to raise a warning in the test suite that is non-breaking, then it should probably use the log function to do so. If this indicates a serious problem or faulty test, we probably should raise an error and fail deadly (even if it only happens after so many attempts.)

We do something very similar to this using `try_wait_until` in `synchronization_helpers.rb`: maybe the HTTP implementation should use this to drive its looping behavior instead.
